scripts/wildcard-domain-processor/wildcard-domains-updater.ts

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable no-await-in-loop,no-restricted-syntax,no-console */
+import * as path from 'path';
+import { RuleParser } from '@adguard/agtree';
+// There is no type definition available for the following import.
+// @ts-ignore
+import { findDeadDomains } from '@adguard/dead-domains-linter/src/urlfilter';
+import { extractRuleDomains } from './domain-extractor';
+import { utils } from './utils';
+import { TOP_LEVEL_DOMAIN_LIST } from './top-tld';
+import { findFilterFiles, readFile, writeFile } from './file-utils';
+
+/**
+ * Parses a rule and extracts domains from it.
+ * @param rule - The rule to extract domains from.
+ * @returns An array of domains extracted from the rule.
+ * @throws Will log an error if the rule cannot be parsed.
+ */
+export function getDomains(rule: string): string[] {
+    let ruleAst;
+    try {
+        ruleAst = RuleParser.parse(rule);
+    } catch (e) {
+        // eslint-disable-next-line no-console
+        console.log(`Unable to parse rule: "${rule}", because of the error: ${e}`);
+        return [];
+    }
+    const domains = extractRuleDomains(ruleAst);
+    return domains;
+}
+
+/**
+ * Extracts wildcard domains from the content of a filter.
+ * @param filterContent - The content of the filter.
+ * @returns A set of wildcard domains extracted from the filter content.
+ */
+function getWildcardDomains(filterContent: string): Set<string> {
+    const rules = filterContent.split(/\r?\n/);
+    const wildcardDomains = new Set<string>();
+    for (const rule of rules) {
+        const domains = getDomains(rule);
+        const wildcardDomainsList = domains.filter((domain) => utils.isWildcardDomain(domain));
+        wildcardDomainsList.forEach((domain) => wildcardDomains.add(domain));
+    }
+    return wildcardDomains;
+}
+
+/**
+ * A map of wildcard domains with all possible TLDs.
+ */
+export type WildcardDomains = { [key: string]: string[] };
+
+/**
+ * Supplements the wildcard domains with all possible TLDs.
+ * @param wildcardDomains - The set of wildcard domains to supplement.
+ * @returns A map of wildcard domains with all possible TLDs.
+ */
+function supplementWithTld(wildcardDomains: Set<string>): WildcardDomains {
+    const wildcardDomainsWithTld: WildcardDomains = {};
+    for (const wildcardDomain of wildcardDomains) {
+        const baseWithoutWildcard = wildcardDomain.slice(0, -2);
+        wildcardDomainsWithTld[wildcardDomain] = [];
+        for (const tld of TOP_LEVEL_DOMAIN_LIST) {
+            wildcardDomainsWithTld[wildcardDomain].push(`${baseWithoutWildcard}.${tld}`);
+        }
+    }
+    return wildcardDomainsWithTld;
+}
+
+/**
+ * Filters out dead domains from a list of domains.
+ * @param value - The list of domains to filter.
+ * @returns A list of alive domains.
+ */
+async function getAliveDomains(value: string[]): Promise<string[]> {
```

Comment:
```suggestion
async function getAliveDomains(rawDomains: string[]): Promise<string[]> {
```

Reply:
Thank you for this suggestion, but please do not create non-working suggestions, as it also requires changing the argument name in the code below.